### PR TITLE
ONEUP-6218: github composite nodejs ECS

### PR DESCRIPTION
**Pull request by yolch-yolchyan**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    default: 'false'
+
+  up-slack-webhook-url:
+    description: 'Slack channel webhook URL used when reporting master build status'
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v2
```
Comment (querwurzel) [inline action.yml]: v3 is latest

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        GIT_COMMIT_MSG_NO_SUFFIX="${GIT_COMMIT_MSG#*\(#}"
+        GIT_PR="${GIT_COMMIT_MSG_NO_SUFFIX%\)*}"
+        GIT_REPO_URL="https://github.com/${{ github.repository }}"
+        GIT_PR_URL="$GIT_REPO_URL/pull/$GIT_PR"
+        GIT_REPO_NAME="${{ github.repository }}"
+        echo "GIT_ACTIONS_URL=<$GIT_REPO_URL/actions/runs/${{ github.run_id }}|${GIT_REPO_NAME#*/}>" >> $GITHUB_ENV
+        [[ "$GIT_PR" =~ ^[0-9]+$ ]] && GIT_PR_MSG="<$GIT_PR_URL|$GIT_COMMIT_MSG>" || GIT_PR_MSG="$GIT_COMMIT_MSG"
+        echo "GIT_PR_MSG=$GIT_PR_MSG" >> $GITHUB_ENV
+    - name: Report - success [on master]
+      uses: Ilshidur/action-slack@2.0.2
```
Comment (querwurzel) [inline action.yml]: v2.1.0 is latest, same for success further down

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        filters: |
+          terraform:
+            - 'terraform/**'
+
+    - name: deploy infrastructure
+      if: steps.infrastructure-changes.outputs.terraform == 'true'
+      uses: spring-media/1up-github-actions@deploy-infrastructure
+      with:
+        service-name: ${{ env.SERVICE_NAME }}
+        pkg-user: ${{ inputs.pkg-user }}
```
Comment (querwurzel) [inline action.yml]: pkg-user is no longer required :)

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        GIT_COMMIT_MSG_NO_SUFFIX="${GIT_COMMIT_MSG#*\(#}"
+        GIT_PR="${GIT_COMMIT_MSG_NO_SUFFIX%\)*}"
+        GIT_REPO_URL="https://github.com/${{ github.repository }}"
+        GIT_PR_URL="$GIT_REPO_URL/pull/$GIT_PR"
+        GIT_REPO_NAME="${{ github.repository }}"
+        echo "GIT_ACTIONS_URL=<$GIT_REPO_URL/actions/runs/${{ github.run_id }}|${GIT_REPO_NAME#*/}>" >> $GITHUB_ENV
+        [[ "$GIT_PR" =~ ^[0-9]+$ ]] && GIT_PR_MSG="<$GIT_PR_URL|$GIT_COMMIT_MSG>" || GIT_PR_MSG="$GIT_COMMIT_MSG"
+        echo "GIT_PR_MSG=$GIT_PR_MSG" >> $GITHUB_ENV
+    - name: Report - success [on master]
+      uses: Ilshidur/action-slack@2.0.2
```
Comment (querwurzel) [inline action.yml]: 2.1.0 here as well ;)

File: action.yml [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    required: true
+
+  docker-image-tag:
+    description: 'Docker image build tag used when pushing to the 1up AWS ECR'
+    required: false
+    default: 'b${{ github.run_number }}-${{ github.sha }}'
+
+  github-token:
+    description: Github token, current token from secrets
+    required: false
```
Comment (querwurzel) [inline action.yml]: Should be required as we just learned :o

File: action.yml [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v3
+
+    - name: Setup - Validate Inputs
+      shell: bash
+      run: |
+        [[ "${{ inputs.pkg-token }}" ]] || { echo "input 'pkg-token' cannot be blank"; exit 1; }
```
Comment (querwurzel) [inline action.yml:47]: You might want to check the github-token here for fast failure

Comment (querwurzel) [reply action.yml:47]: pkg-user being back?

File: action.yml (hunk 3 above)
Comment (querwurzel) [reply action.yml]: pkg-user being back?

File: action.yml [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+        GIT_COMMIT_MSG=$(git log -1 --pretty=format:'%s')
+        GIT_COMMIT_MSG_NO_SUFFIX="${GIT_COMMIT_MSG#*\(#}"
+        GIT_PR="${GIT_COMMIT_MSG_NO_SUFFIX%\)*}"
+        GIT_REPO_URL="https://github.com/${{ github.repository }}"
+        GIT_PR_URL="$GIT_REPO_URL/pull/$GIT_PR"
+        GIT_REPO_NAME="${{ github.repository }}"
+        echo "GIT_ACTIONS_URL=<$GIT_REPO_URL/actions/runs/${{ github.run_id }}|${GIT_REPO_NAME#*/}>" >> $GITHUB_ENV
+        [[ "$GIT_PR" =~ ^[0-9]+$ ]] && GIT_PR_MSG="<$GIT_PR_URL|$GIT_COMMIT_MSG>" || GIT_PR_MSG="$GIT_COMMIT_MSG"
+        echo "GIT_PR_MSG=$GIT_PR_MSG" >> $GITHUB_ENV
+      shell: bash
```
Comment (querwurzel) [inline action.yml:142]: Are you fine with moving shell: bash before the bash script? This applies to all steps using shell scripts :)

File: action.yml (hunk 3 above)
Comment (yolch-yolchyan) [reply action.yml]: it was there since beginning, I'm still testing this pipeline

Comment (yolch-yolchyan) [reply action.yml]: not working with main one

File: action.yml (hunk 7 above)
Comment (yolch-yolchyan) [reply action.yml:142]: what's the difference?

Comment (querwurzel) [reply action.yml:142]: Technically none, but in my regards a bit in terms of readability

File: action.yml (hunk 3 above)
Comment (querwurzel) [reply action.yml]: Just merged it :)

File: action.yml (hunk 7 above)
Comment (yolch-yolchyan) [reply action.yml:142]: maybe difference in thinking, I feel at the end if more readable :D

Comment (yolch-yolchyan) [reply action.yml:142]: at least can find out easily where this is not added

**Review by querwurzel — APPROVED** [review]
LGTM

**Pull request merged**